core/src/main/java/io/questdb/cairo/pool/ReaderPool.java

```diff
@@ -346,7 +350,35 @@ public static class Entry {
         public Entry(int index, long currentMicros) {
             this.index = index;
             Arrays.fill(allocations, UNALLOCATED);
-            Arrays.fill(releaseTimes, currentMicros);
+            Arrays.fill(releaseOrAcquireTimes, currentMicros);
+        }
+
+        public long getOwnerVolatile(int pos) {
+            assert pos >= 0 && pos <= ENTRY_SIZE;
+            return Unsafe.arrayGetVolatile(allocations, pos);
+        }
+
+        public long getReleaseOrAcquireTimeVolatile(int pos) {
+            assert pos >= 0 && pos <= ENTRY_SIZE;
+            return Unsafe.arrayGetVolatile(releaseOrAcquireTimes, pos);
+        }
+
+        public R getReader(int pos) {
+            assert pos >= 0 && pos <= ENTRY_SIZE;
```

Comment:
Perhaps we don't need it. It will throw `ArrayIndexOutOfBoundsException` anyway. 

The explicit assert helps a reader a bit, but given we have assertions enabled in the product it's probably not justifiable.  I will remove it.